### DISCUSSION TITLE
How to modify EEPROM value

### DISCUSSION THREAD

**Discussion by CHNtentes** [OE4T/meta-tegra#1171] [Q&A] [answered]
Hi. I'm using Orin NX 16GB, but with a custom carrier board, which has no EEPROM on it.
As a result, I need to follow NV's instruction to change "cvb_eeprom_read_size" to 0x0.
It's from here: [https://docs.nvidia.com/jetson/archives/r35.2.1/DeveloperGuide/text/HR/JetsonModuleAdaptationAndBringUp/JetsonOrinNxSeries.html#eeprom-modifications](url)
How can I make the same change with meta-tegra?

**Comment by madisongh** [comment, marked as answer]
You would take the same basic approach as mentioned on [the wiki page](https://github.com/OE4T/meta-tegra/wiki/Creating-a-custom-MACHINE) for defining a custom machine based on the TX2 or Xavier:
* Create a recipe that installs your customized bootloader configuration file(s), adding a bbappend for the `tegra-bootfiles` recipes so it depends on your added recipe
* Add a bbappend to the `tegra-flashvars` recipe so it can pick up the custom `flashvars` file for your custom machine.

**Reply by CHNtentes** [reply]
Thanks for reply. I found the dts file which I need to modify under build/tmp/work-shared/L4T-native-35.2.1-r0. So it's extracted from NV original BSP package?

**Reply by madisongh** [reply]
Correct

**Reply by StefanJV** [reply]
Hi Madison. I'm on the styhead branch and I'm trying to make this same change work.
I created a recipe that installs the custom dts file to ${D}${datadir}/tegraflash. I added the bbappend for tegra-bootfiles to make it depend on my new recipe.
I changed the TEGRA_FLASHVAR_MB2BCT_CFG variable in my machine.conf, and that seems to work (in terms of the resulting flashvars file).
But I can't get the file to show up in the tegraflash folder.
What step am I missing? Do I need to do an install:append in the tegra-bootfiles recipe?

**Reply by madisongh** [reply]
It depends on how you named your file.  [These commands](https://github.com/OE4T/meta-tegra/blob/2bbc15ae8b5aa1dc570eae544cd15d83ce239a9a/classes-recipe/image_types_tegra.bbclass#L284-L289) copy the boot files into the tegraflash package from the `/usr/share/tegraflash` directory of the recipe sysroot. If you named your custom `.dts` file without the `tegra234-` prefix, it wouldn't get covered by those commands.

**Reply by StefanJV** [reply]
Matt, Thanks! Prepending tegra234- to my dts name solved the issue.
I'm okay with that as a solution, but would like to get a better understanding.
I was having trouble figuring out how that copy was being done, so thanks for the link to the bbclass.
So, it seems like this is an image recipe for tegraflash in the bbclass file. Is there a way to append to the tegraflash_populate_package function to pick up the file without renaming?

**Reply by madisongh** [reply]
Yes, `tegraflash` is an image type, and that bbclass is mainly used by image recipes. There are some hooks in there for customizing its behavior. The one that you would use for this purpose is [tegraflash_custom_pre](https://github.com/OE4T/meta-tegra/blob/2bbc15ae8b5aa1dc570eae544cd15d83ce239a9a/classes-recipe/image_types_tegra.bbclass#L108-L115). You would implement that function in your image recipe to copy in the additional file(s).

**Reply by StefanJV** [reply]
Awesome. Thanks very much for the help and all the work on this. It is much appreciated.

**Comment by CHNtentes** [comment]
Thanks. I've got another problem here. I bitbaked "tegra-initrd-flash-initramfs" without error, but there's no tegraflash.tar.gz file under build/tmp/deploy/images. What could be wrong? Or is this an expected behaviour...

Here's what is currently in that directory:
[deploy-images.txt](https://github.com/OE4T/meta-tegra/files/10727668/deploy-images.txt)

And here's my local.conf:
[local.conf.txt](https://github.com/OE4T/meta-tegra/files/10727672/local.conf.txt)

**Reply by CHNtentes** [reply]
I figured it out. tegra-initrd-flash-initramfs.bb contains IMAGE_FSTYPES = "${TEGRA_INITRD_FLASH_INITRAMFS_FSTYPES}", which overwrites tegraflash FSTYPE.

**Reply by madisongh** [reply]
Just build a regular image, and the initrd-flash kernel/initrd will be built and bundled into the tegraflash package for you.  Initramfs recipes are not normal images, so it doesn't make sense to do tegraflash packages for them.
